bench/meili/mapmatch.cc

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <sstream>
+
+#include <benchmark/benchmark.h>
+#include <boost/property_tree/ptree.hpp>
+#include <rapidjson/rapidjson.h>
+
+#include "meili/map_matcher_factory.h"
+#include "meili/measurement.h"
+#include "sif/costconstants.h"
+#include "sif/costfactory.h"
+#include "tyr/actor.h"
+
+using namespace valhalla::midgard;
+using namespace valhalla::meili;
+using namespace valhalla::sif;
+
+namespace {
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+constexpr float kGpsAccuracy = 4.07;
+constexpr float kSearchRadius = 50;
+
+// Inline benchmarks for OfflineMatch
+
+class OfflineMapmatchFixture : public benchmark::Fixture {
+public:
+  void SetUp(const ::benchmark::State& state) {
+    InitEngineConfig();
+    InitMapMatcher();
+  }
+
+  void TearDown(const ::benchmark::State& state) {
+    matcher_factory_->ClearCache();
+    mapmatcher_->Clear();
+    matcher_factory_.reset();
+    mapmatcher_.reset();
+  }
+
+private:
+  void InitEngineConfig() {
+    rapidjson::read_json(VALHALLA_SOURCE_DIR "bench/meili/config.json", config_);
+    const rapidjson::Document doc;
+    ParseAutoCostOptions(doc, "/costing_options/auto", options_.add_costing_options());
+  }
+
+  void InitMapMatcher() {
+    matcher_factory_ = std::make_shared<MapMatcherFactory>(config_);
+    mapmatcher_.reset(matcher_factory_->Create(valhalla::Costing::auto_, options_));
+  }
+
+protected:
+  // Mapmatcher implementation
+  std::shared_ptr<MapMatcher> mapmatcher_;
+  std::shared_ptr<MapMatcherFactory> matcher_factory_;
```

Comment:
could we use the `tyr::actor_t` interface here and actually pass requests to the map matcher? this would be a more  holistic test in that way. as it stands this only tests the performance of the map matching (routing and candidate search) but misses all of the transformations that need to be done on that output to use it for building a route with directions.

Reply:
totes, @kevinkreiser. That end-to-end case is actually covered by the very next benchmark: https://github.com/valhalla/valhalla/pull/2222/files#diff-920e78dd18a33546d72bcbe3acf6327aR103.

We *also* want to have a standalone microbenchmark on the map matching sans serialization, right?

Reply:
sounds good to me, sorry i completely missed that below!